cmd/cosign/cli/attest/attest_blob.go

```diff
@@ -258,3 +281,41 @@ func (c *AttestBlobCommand) Exec(ctx context.Context, artifactPath string) error
 
 	return nil
 }
+
+func makeNewBundle(rekorEntry *models.LogEntryAnon, payload, sig, signer, timestampBytes []byte, payloadType string) ([]byte, error) {
+	// Determine if signature is certificate or not
+	var hint string
+	var rawCert []byte
+
+	cert, err := cryptoutils.UnmarshalCertificatesFromPEM(signer)
+	if err != nil || len(cert) == 0 {
+		hashedBytes := sha256.Sum256(signer)
```

Comment:
`signer` is a [PEM-encoded](https://github.com/sigstore/cosign/blob/f7a5725317fbb0ad3317ceddc345d21cc85151e4/cmd/cosign/cli/sign/sign.go#L606) when not a cert. If we want to align to our suggested key hint, which is [`SHA-256(DER(SPKI(key)))`](https://www.rfc-editor.org/rfc/rfc6962#section-3.2:~:text=represented%20as%20SubjectPublicKeyInfo.-,%22issuer_key_hash%22,-is%20the%20SHA), we'll need to do:

```
pkixPubKey, err := x509.MarshalPKIXPublicKey(signer.PublicKey())
if err != nil {
  return nil, err
}
hashedBytes := sha256.Sum256(pkixPubKey)
hint = base64.StdEncoding.EncodeToString(hashedBytes[:])
```

Reply:
And same for sign-blob